src/main/java/net/dv8tion/jda/internal/requests/ratelimit/BotRateLimiter.java

```diff
@@ -102,6 +103,33 @@ private ScheduledExecutorService getScheduler()
         return requester.getJDA().getRateLimitPool();
     }
 
+    @Override
+    public int cancelRequests()
+    {
+        return MiscUtil.locked(bucketLock, () -> {
+            for (Future<?> future : rateLimitQueue.values())
+                future.cancel(false);
+            rateLimitQueue.clear();
+
+            AtomicInteger count = new AtomicInteger(0);
+            buckets.values()
+                .stream()
+                .map(Bucket::getRequests)
+                .flatMap(Collection::stream)
+                .filter(request -> !request.isPriority())
+                .forEach(request -> {
+                    request.cancel();
+                    count.incrementAndGet();
+                });
+            buckets.clear();
```

Comment:
Do we want to clear priority requests? Based on the filter above, we don't cancel priority requests, but this will clear all request out regardless.

Reply:
Hmm, will need to think how to handle this nicely.